server/src/main/java/org/elasticsearch/index/store/Store.java

```diff
@@ -1295,6 +1303,19 @@ public void seek(long pos) throws IOException {
             }
         }
 
+        @Override
+        public void skipBytes(long numBytes) throws IOException {
+            if (skipBuffer == null) {
+                skipBuffer = new byte[SKIP_BUFFER_SIZE];
+            }
+            assert skipBuffer.length == SKIP_BUFFER_SIZE;
+            for (long skipped = 0; skipped < numBytes; ) {
+                final int step = (int) Math.min(SKIP_BUFFER_SIZE, numBytes - skipped);
+                readBytes(skipBuffer, 0, step);
+                skipped += step;
+            }
+        }
```

Comment:
It looks buggy to me that this implementation is not taking into account `verifiedPosition`?

Reply:
I think we might be able to do something simpler here by leaving `skipBytes` unimplemented (so that it uses the implementation from the parent class, which calls seek) and changing `seek` to use `super.seek` rather than `skipBytes` to skip over some bytes.

Reply:
++, I've done this in 126f0c05b00e787665fbb6669903f0509eb44a37